### DISCUSSION TITLE
Moving from zwave-js to zwave-js-ui - Fedora 40, Portainer Docker -  latest zwave-js-ui image.

### DISCUSSION THREAD

**Discussion by robertwynkoop** [zwave-js/zwave-js-ui#3861] [General]
Fedora Portainer, I'm not able to get the environment variable to be used for keys.  The js-ui gives me an error about keys.

I was able to enter the keys in in the zwave-js-ui an get connected to my USB stick.  However I think I need environment vars to work since I want to be able to update easily. 

With the container running, I can access the terminal and issue a set command to see the environments are set.  However, zwave-js-ui does not seem to pick up the values.

Additionally, even though I have port 3000 exposed, I can not get my home assistant instance to connect.

**Comment by alexojegu** [comment]
> I was able to enter the keys in in the zwave-js-ui an get connected to my USB stick. However I think I need environment vars to work since I want to be able to update easily.

It's better to use a Docker volume (or a bind mount) to permanently store all the configuration (I don't know how to do this with Portainer) https://zwave-js.github.io/zwave-js-ui/#/getting-started/docker

> With the container running, I can access the terminal and issue a set command to see the environments are set. However, zwave-js-ui does not seem to pick up the values.

Are you using the correct environment variables? https://zwave-js.github.io/zwave-js-ui/#/guide/env-vars

> Additionally, even though I have port 3000 exposed, I can not get my home assistant instance to connect.

Are Home Assistant and Z-Wave JS UI on the same network? If not, can Home Assistant communicate with Z-Wave JS UI? Do you have "Home Assistant" -> "WS Server" enabled in Z-Wave JS UI settings?

**Reply by robertwynkoop** [reply]
Humm, Ok I used a bind mount to a local directory.  It works, I can see logs and setting being saved.

Yes, I was using the correct keys but they were not being used.   Could it be something with the entry point / container startup?
CMD | node server/bin/www
ENTRYPOINT | null

It does look like Home Assistant is picking up devices after zwave-js-ui lists them as known / and no longer unknown mfg / model....  Not sure what is going on here.  Prior attempts would error out connecting to the 3000 service port. Maybe I was not patient enough?  But it is working now.

So, I think I'm working.  I will let the device control panel rebuild and see where I'm at later in the morning.

Robert Wynkoop

**Comment by robertsLando** [comment]
Sorry for late reply but I was on vacation, is this fixed now?

**Reply by robertwynkoop** [reply]
Yes, I have zwave-js-ui latest running in Fedora 40 Portainer(Docker) now.  I had bind a local config dir location.  Not sure why my environment vars were not being used however.  It could be that they were set after the js-ui had already started running in the container would be my guess.

The configured entry cmd and entry point are:
CMD | node server/bin/www
ENTRYPOINT | null

Not sure if that is right or not.

But I am working now.  So now I back to fighting zwave range issues with a new 30x50 shop.

Robert Wynkoop
